.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,13 +1,23 @@
 Fixes #
+
 Design discussion issue #
 
 ## Changes
 
-Please provide a brief description of the changes here.
+Please provide a brief description of the changes here. If further adjustments
+are made following feedback or additional insights, please update this
+accordingly to maintain an accurate record of the change.
 
 ## Merge requirement checklist
 
-* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
+* [ ]
+  [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md)
+  guidelines followed (license requirements, nullable enabled, static analysis,
+  etc.)
 * [ ] Unit tests added/updated
-* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
+* [ ] For all non-trivial modifications, please ensure the `CHANGELOG.md` file is
+  duly updated. Each changelog entry should be self-explanatory, providing a
+  concise, yet comprehensive overview of the changes made. In instances where a
+  change is too extensive for a brief changelog description, detail the
+  modifications within the pull request description and add link to it.
```

Comment:
I get what you are going for, but I feel like this is a lot of text to have in the template. Could we link to a section in contributing guide which has more details for users who need help?

Reply:
Thats what I was thinking initially... But then thought that not many people read the contributing guide, but PR template is harder to ignore....

Reply:
IMO link(s) to PR should be always added

Reply:
Agree to "Link to PR is always needed". Will modify wording.

Reply:
> Thats what I was thinking initially... But then thought that not many people read the contributing guide, but PR template is harder to ignore....

IMO it is too spammy and it is going to be annoying to the regular contributors and people reviewing PRs. It should be short here. If people don't follow it, that is fine. Part of the approver/maintainer workflow when reviewing PRs is to verify it is all followed. Having it in contributing would help those who read it, but also gives us something we can link people when they don't follow it (or do it wrong) 🤣 

My suggestion is to do like...

```md
* [ ] [Appropriate CHANGELOG files updated for non-trivial changes](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md#changelog-modification-guidelines)
```

Not a blocker for me, but no guarantee I won't try to change it in the future if it becomes annoying 😄

Reply:
no, I think this is more apt. Let me update accordingly.